lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -282,25 +288,33 @@ def create_or_update_group(connection, module):
 
         if purge_users:
             for user in list(set(current_group_members_list) - set(users)):
-                try:
-                    connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
-                except ClientError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc(),
-                                     **camel_dict_to_snake_dict(e.response))
-                except ParamValidationError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc())
-
-        if users != [None]:
-            for user in users:
-                try:
-                    connection.add_user_to_group(GroupName=params['GroupName'], UserName=user)
-                except ClientError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc(),
-                                     **camel_dict_to_snake_dict(e.response))
-                except ParamValidationError as e:
-                    module.fail_json(msg=e.message, exception=traceback.format_exc())
-
-        changed = True
+                # Ensure we mark things have changed if any user gets purged
+                changed = True
+                # Skip actions for check mode
+                if not module.check_mode:
+                    try:
+                        connection.remove_user_from_group(GroupName=params['GroupName'], UserName=user)
+                    except ClientError as e:
+                        module.fail_json(msg=e.message, exception=traceback.format_exc(),
+                                         **camel_dict_to_snake_dict(e.response))
+                    except ParamValidationError as e:
+                        module.fail_json(msg=e.message, exception=traceback.format_exc())
+        # If there are users to adjust that aren't in the current list, then things have changed
+        # Otherwise the only changes were in purging above
+        if set(users).difference(set(current_group_members_list)):
```

Comment:
The above block of code handles when the `current_group_members_list` contains entries that do not exist in our desired `users` list. Running `set(users).difference()` will result in a list of what users we desire that are not already present. The lower block doesn't need to run at all if you're not adding any users to the group. If we used the simple equality check it'd still run.